src/path.jl

```diff
@@ -533,8 +533,13 @@ julia> created(p"src/FilePathsBase.jl")
 ```
 """
 created(fp::AbstractPath) = stat(fp).ctime
-Base.isdir(fp::AbstractPath) = isdir(mode(fp))
+
+Base.isdir(fp::AbstractPath) = isdir(directoriestype(fp), fp)
+Base.isdir(::DirectoriesExplicit, fp::AbstractPath) = isdir(mode(fp))
+
 Base.isfile(fp::AbstractPath) = isfile(mode(fp))
+
+#TODO: most of these clearly only make sense on local FS, should they have traits?
```

Comment:
I think having a trait for this is a very good idea if the primary goal is to more cleanly fail for unsupported functionality. This could still get weird in the sense that S3Paths do still have permissions, but they work differently.